### PR TITLE
net/gcoap: fix sock send return type

### PR DESCRIPTION
**Pull request by kb2ma**
### Contribution description
During the review for #7337, I found that gcoap assumed the return type for gcoap_udp_send() was of type `size_t`, but really it is of type `ssize_t`. This PR fixes that error. I also added a debug statement to show the error code from an invocation of gcoap_udp_send().

### Issues/PRs references
[Discussion](https://github.com/RIOT-OS/RIOT/pull/7337#issuecomment-357868126) in the earlier PR.

### CODE REVIEW

File: sys/net/application_layer/gcoap/gcoap.c [hunk 1]
```diff
@@ -832,14 +835,14 @@ size_t gcoap_req_send2(const uint8_t *buf, size_t len,
...
-    if (!res) {
+    if (res <= 0) {
         if (msg_type == COAP_TYPE_CON) {
             *memo->msg.data.pdu_buf = 0;    /* clear resend buffer */
         }
         memo->state = GCOAP_MEMO_UNUSED;
         DEBUG("gcoap: sock send failed: %d\n", (int)res);
     }
-    return res;
+    return (size_t)(res > 0 ? res : 0);
```
Comment (miri64) [inline sys/net/application_layer/gcoap/gcoap.c]: Please add some parenthesis around `res > 0`.

File: sys/net/application_layer/gcoap/gcoap.c [hunk 2]
```diff
@@ -902,7 +905,8 @@ size_t gcoap_obs_send(const uint8_t *buf, size_t len,
     _find_obs_memo_resource(&memo, resource);
 
     if (memo) {
-        return sock_udp_send(&_sock, buf, len, memo->observer);
+        ssize_t bytes = sock_udp_send(&_sock, buf, len, memo->observer);
+        return (size_t)(bytes > 0 ? bytes : 0);
```
Comment (miri64) [inline sys/net/application_layer/gcoap/gcoap.c]: Please add some parenthesis around `bytes > 0`.

**Comment by kb2ma** [comment]
Thanks for the quick review, Martine. I went ahead and squashed in the improvements.

**Comment by miri64** [comment]
(Including `gnrc_neterr` let's `sock` know, that the host is unreachable via [this line](https://github.com/RIOT-OS/RIOT/blob/master/sys/net/gnrc/network_layer/ipv6/nib/nib.c#L203))

**Pull request merged**